xbmc/guilib/guiinfo/MusicGUIInfo.cpp

```diff
@@ -99,16 +100,25 @@ bool CMusicGUIInfo::GetLabel(std::string& value, const CFileItem *item, int cont
         return true;
       case PLAYER_TITLE:
         value = tag->GetTitle();
+        if (!tag->GetStationName().empty() && tag->GetTitle().empty())
+        {
+          value.clear();
+          return true;
+        }
         return !value.empty();
       case MUSICPLAYER_TITLE:
         value = tag->GetTitle();
         if (value.empty())
           value = item->GetLabel();
         if (value.empty())
           value = CUtil::GetTitleFromPath(item->GetPath());
+        if (!tag->GetStationName().empty() && tag->GetTitle().empty())
+          value.clear();
         return true;
       case LISTITEM_TITLE:
         value = tag->GetTitle();
+        if (!tag->GetStationName().empty() && tag->GetTitle().empty())
+          value.clear();
```

Comment:
see my other comment. `value` is already empty here. This line can be removed. Thus, the whole check can be removed, too.

Reply:
I don't think that for any other music use-case that this value should ever be empty. However, I'd rather have a robust check in place so as to guarantee the label is only blank in one scenario than leave it to chance that it's empty.  Users do strange things with their music libraries at times.  Checking both values seems like a sane thing to do to me but maybe I'm wrong?  @DaveTBlake Do you have any thoughts on this?

Reply:
Not sure I see your logic @the-black-eagle, step through it.
Say we have a station name, if  tag->GetTitle() is empty, then value is empty already. If tag->GetTitle() is not empty, then value is not empty and is not cleared. 
Say we do not have a station name, if  tag->GetTitle() is empty, then value is empty already. If tag->GetTitle() is not empty, then value is not empty and is not cleared. 
I don't see why this is needed no matter what _strange things_ users do.

Reply:
Gotcha. Must have got twisted up in my own strange logic!